### PR TITLE
librealsense: add enableGUI switch 

### LINKED ISSUE #165140: Packaging request: realsense-viewer
**Project description**

Realsense-viewer is an official application for playing around with Intel® RealSense™ depth cameras (D400 series and the SR300).
We already have [librealsense](https://github.com/NixOS/nixpkgs/blob/nixos-21.11/pkgs/development/libraries/librealsense/default.nix#L72) in nixpkgs, but it does not include all of the official tools:

```
$ ls /nix/store/sgcm8np2ix9vjal8safbwc773b5imwid-librealsense-2.45.0/bin 
rs-color    rs-distance           rs-fw-update        rs-pose-and-image  rs-save-to-disk
rs-convert  rs-enumerate-devices  rs-hello-realsense  rs-pose-predict    rs-terminal
rs-depth    rs-fw-logger          rs-pose             rs-record
```

**Metadata**

* homepage URL: https://github.com/IntelRealSense/librealsense/tree/master/tools/realsense-viewer
* source URL: https://github.com/IntelRealSense/librealsense/tree/master/tools/realsense-viewer 
* license: asl20
* platforms: unix

@brian-dawn :pray:

### PR DESCRIPTION
**Pull request by pbsds**
###### Description of changes

This will compile the realsense-viewer application.
Fixes https://github.com/NixOS/nixpkgs/issues/165140

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

~~Not yet tested with relevant hardware, i need to head to the office to do so.
Draft PR in the meanwhile.~~

Tested

Subject to bikeshedding: What should the flag be named? I think `enableGUI` is not *that* discoverable...

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -9848,6 +9848,12 @@
     githubId = 6498458;
     name = "pebble kite";
   };
+  pbsds = {
```
Comment (brian-dawn) [inline maintainers/maintainer-list.nix:9851]: I see you're adding yourself to the maintainers list, do you also want to add yourself as a maintainer of the librealsense package? I certainly would appreciate others being able to test changes! :)

**Review by brian-dawn — APPROVED** [review]
This looks good to me. I also won't be able to test this with real hardware as I would need to head into the office which I won't be able to do for some time.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (pbsds) [reply maintainers/maintainer-list.nix:9851]: Whoopsy, forgot that step

**Review by brian-dawn — APPROVED** [review]

**Comment by pbsds** [comment]
Works.
Screenshot from 2022-06-01 00-37-42

**Review by brian-dawn — APPROVED** [review]

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 175048` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>librealsense (librealsenseWithoutCuda)</li>
    <li>librealsenseWithCuda</li>
    <li>librealsenseWithGUI</li>
    <li>python310Packages.pyrealsense2 (python310Packages.pyrealsense2WithoutCuda)</li>
    <li>python310Packages.pyrealsense2WithCuda</li>
    <li>python39Packages.pyrealsense2 (python39Packages.pyrealsense2WithoutCuda)</li>
    <li>python39Packages.pyrealsense2WithCuda</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -23319,6 +23319,10 @@ with pkgs;
     cudaSupport = false;
   };
 
+  librealsenseWithGUI = callPackage ../development/libraries/librealsense {
```
Comment (fufexan) [inline pkgs/top-level/all-packages.nix:23322]:
```suggestion
  librealsense-gui = callPackage ../development/libraries/librealsense {
```
looks better to me

**Review by fufexan — COMMENTED** [review]
Looks fine to me, apart from the name.

File: pkgs/development/libraries/librealsense/default.nix [hunk 3]
```diff
@@ -32,7 +38,8 @@ stdenv.mkDerivation rec {
     libusb1
     gcc.cc.lib
   ] ++ lib.optional cudaSupport cudaPackages.cudatoolkit
-    ++ lib.optionals enablePython (with pythonPackages; [python pybind11 ]);
+    ++ lib.optionals enablePython (with pythonPackages; [python pybind11 ])
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/librealsense/default.nix:41]:
```suggestion
    ++ lib.optionals enablePython (with pythonPackages; [ python pybind11 ])
```

**Review by fufexan — APPROVED** [review]

**Pull request merged**